File: app/src/main/java/com/warunya/ricenearby/base/BaseView.java
package com.warunya.ricenearby.base;

public interface BaseView {

    void showProgress();

    void hideProgress();

    void showNotFound();

    void hideNotFound();

    interface Progress {
        void updateProgress(String message);

        void showProgressDialog();

        void hideProgressDialog();

        void error(String message);
    }
}

File: app/src/main/java/com/warunya/ricenearby/base/BaseView.kt
package com.warunya.ricenearby.base

interface BaseView {
    fun showProgress()
    fun hideProgress()
    fun showNotFound()
    fun hideNotFound()

    interface Progress {
        fun updateProgress(message: String)
        fun showProgressDialog()
        fun hideProgressDialog()
        fun error(message: String)
    }
}
